packages/suite/src/views/wallet/transactions/TransactionList/TransactionListActions/TransactionListActions.tsx

```diff
@@ -19,20 +40,89 @@ export const TransactionListActions = ({
     setSelectedPage,
     accountMetadata,
     isExportable = true,
-}: TransactionListActionsProps) => (
-    <Wrapper>
-        <SearchAction
-            account={account}
-            searchQuery={searchQuery}
-            setSearch={setSearch}
-            setSelectedPage={setSelectedPage}
-        />
-        {isExportable && (
-            <ExportAction
-                account={account}
-                searchQuery={searchQuery}
-                accountMetadata={accountMetadata}
-            />
-        )}
-    </Wrapper>
-);
+}: TransactionListActionsProps) => {
+    const [isExpanded, setExpanded] = useState(false);
+    const [hasFetchedAll, setHasFetchedAll] = useState(false);
+
+    const inputRef = useRef<HTMLInputElement | null>(null);
+    const dispatch = useDispatch();
+    const { translationString } = useTranslation();
+
+    const onSearch = useCallback(
+        async ({ target }: ChangeEvent<HTMLInputElement>) => {
+            setSelectedPage(1);
+            setSearch(target.value);
+
+            if (!hasFetchedAll) {
+                setHasFetchedAll(true);
+
+                try {
+                    await dispatch(
+                        fetchAllTransactionsForAccountThunk({
+                            accountKey: account.key,
+                            noLoading: true,
+                        }),
+                    );
+                } catch (err) {
+                    dispatch(
+                        notificationsActions.addToast({
+                            type: 'error',
+                            error: translationString('TR_SEARCH_FAIL'),
+                        }),
+                    );
+                }
+            }
+        },
+        [account, dispatch, hasFetchedAll, setSearch, setSelectedPage, translationString],
+    );
+
+    const onSearchKeys = useCallback(
+        (event: KeyboardEvent) => {
+            if (
+                inputRef.current &&
+                (event.ctrlKey || event.metaKey) &&
+                event.key === KEYBOARD_CODE.LETTER_F
+            ) {
+                event.preventDefault();
+                inputRef.current.focus();
+            }
+        },
+        [inputRef],
+    );
```

Comment:
please noooo, i hate this on tx history list

Reply:
So should I remove it from both places?

Reply:
I think it makes sense in desktop app but I would remove it from web as it blocks native browser search.